### PR TITLE
feat(base-ui): support IDialogService & IConfirmService 

### LINKED ISSUE #117: Basic UI components & services
Implements basic components and services to improve code reuse and provide consistent behavior across different plugins and platforms.

> Deadline Oct 14, 2023

## Components

- [x] `Message`
- [x] `Button`
- [x] `Icon`: integrate `@univerjs/icon`
- [x] `Input` @Dushusir 
- [x] `InputNumber` @Dushusir 
- [x] `Container`
- [x] `Layout`
- [x] `Slider`
- [x] `Tooltip`

## Docs

- [x] Design Token / Palette
- [x] Deploying storybook

## Services

Services should be platform-agnostic.

- [x] ThemeService (manages themes and dark mode)
- [x] `IMessageService`

---

> Deadline Oct 27, 2023

## Components

- [x] `Dropdown`
- [x] `CanvasIcon`: rename `CanvasIcon` to `BorderLine`
- [x] `Menu`
  - [x] ~~`Item`~~
  - [x] `NeoCustomLabel`
  - [x] `CustomLabel`: completely deprecate the`CustomLabel` component
  - [x] ~~`Select`~~
- [x] `ColorPicker`
- [x] `ColorPickerCircleButton`

---

> Deadline Nov 4, 2023

## Components

- [x] ~~`Tabs`~~
- [x] `Dialog`: rename `Modal` to `Dialog`
- [x] `Notification` @Dushusir 
- [x] `Confirm`

## Services

- [x] `INotificationService` @Dushusir 
- [x] `IDialogService`
- [x] `IConfirmService`

---

> Deadline Nov 11, 2023

## Components

- [x] ~~`CellRange`~~
- [x] ~~`Collapse`~~
- [x] ~~`Separator`~~
- [x] ~~`ResizeDialog`~~
- [x] ~~`SiderModal`~~
- [x] `Avatar`
- [x] `Checkbox`
  - [x] `CheckboxGroup`
- [x] `Radio`
  - [x] `RadioGroup`
- [x] ~~`Drag`~~
- [x] ~~`Slot`~~
- [x] ~~`Modal`~~

---

> Not in plan

## Components

- [x] ~~`DatePicker`~~ move to #378

### PR DESCRIPTION
**Pull request by jikkai**
close #117

### CODE REVIEW

File: packages/base-ui/src/services/confirm/confirm.service.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { type Subject } from 'rxjs';
+
+import { type IConfirmPartMethodOptions } from '../../views/components/confirm-part/interface';
+
+export const IConfirmService = createIdentifier<IConfirmService>('univer.confirm-service');
+
+export interface IConfirmService {
+    open(params: IConfirmPartMethodOptions): IDisposable;
+
+    close(): void;
```
Comment (wzhudev) [inline packages/base-ui/src/services/confirm/confirm.service.ts:11]: Is it more appropriate adding a parameter here to close a specific confirm dialog?

File: packages/base-ui/src/services/confirm/desktop-confirm.service.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { toDisposable } from '@univerjs/core';
+import { IDisposable } from '@wendellhu/redi';
+import { Subject } from 'rxjs';
+
+import { type IConfirmPartMethodOptions } from '../../views/components/confirm-part/interface';
+import { IConfirmService } from './confirm.service';
+
+export class DesktopConfirmService implements IConfirmService {
```
Comment (wzhudev) [inline packages/base-ui/src/services/confirm/desktop-confirm.service.ts:8]: The service seems to be designed on an assumption that there is only one confirm dialog at a time. The assumption may need to be re-evaluated.

File: packages/base-ui/src/services/dialog/desktop-dialog.service.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { type IDialogPartMethodOptions } from '../../views/components/dialog-part/interface';
+import { IDialogService } from './dialog.service';
+
+export class DesktopDialogService implements IDialogService {
+    private readonly dialog$ = new Subject<IDialogPartMethodOptions>();
+
+    open(options: IDialogPartMethodOptions): IDisposable {
+        this.dialog$.next(options);
+
+        return toDisposable(() => {});
```
Comment (wzhudev) [inline packages/base-ui/src/services/dialog/desktop-dialog.service.ts:14]: The disposable function does nothing.

**Review by wzhudev — CHANGES_REQUESTED** [review]

File: packages/base-ui/src/services/dialog/dialog.service.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { type Subject } from 'rxjs';
+
+import { type IDialogPartMethodOptions } from '../../views/components/dialog-part/interface';
+
+export const IDialogService = createIdentifier<IDialogService>('univer.dialog-service');
+
+export interface IDialogService {
+    open(params: IDialogPartMethodOptions): IDisposable;
+
+    close(id: string): void;
```
Comment (wzhudev) [inline packages/base-ui/src/services/dialog/dialog.service.ts:11]: Some JSDoc would be nice.

**Review by wzhudev — APPROVED** [review]
LGTM. Could you please add some documents?

File: packages/base-ui/src/services/dialog/dialog.service.ts (hunk 4 above)
Comment (jikkai) [reply packages/base-ui/src/services/dialog/dialog.service.ts:11]: The JSDoc may not fully describe the usage of this method, so I will provide a comprehensive usage example later :octocat:

**Pull request merged**